extensions/sql-database-projects/src/models/options/deployOptionsModel.ts

```diff
@@ -8,250 +8,64 @@ import * as mssql from 'mssql';
-			, this.deploymentOptions.ignoreDmlTriggerOrder.displayName
-			, this.deploymentOptions.ignoreDefaultSchema.displayName
-			, this.deploymentOptions.ignoreDdlTriggerState.displayName
-			, this.deploymentOptions.ignoreDdlTriggerOrder.displayName
-			, this.deploymentOptions.ignoreCryptographicProviderFilePath.displayName
-			, this.deploymentOptions.verifyDeployment.displayName
-			, this.deploymentOptions.ignoreComments.displayName
-			, this.deploymentOptions.ignoreColumnCollation.displayName
-			, this.deploymentOptions.ignoreAuthorizer.displayName
-			, this.deploymentOptions.ignoreAnsiNulls.displayName
-			, this.deploymentOptions.generateSmartDefaults.displayName
-			, this.deploymentOptions.dropStatisticsNotInSource.displayName
-			, this.deploymentOptions.dropRoleMembersNotInSource.displayName
-			, this.deploymentOptions.dropPermissionsNotInSource.displayName
-			, this.deploymentOptions.dropObjectsNotInSource.displayName
-			, this.deploymentOptions.ignoreColumnOrder.displayName
-			, this.deploymentOptions.ignoreTablePartitionOptions.displayName
-			, this.deploymentOptions.doNotEvaluateSqlCmdVariables.displayName
-			, this.deploymentOptions.disableParallelismForEnablingIndexes.displayName
-			, this.deploymentOptions.disableIndexesForDataPhase.displayName
-			, this.deploymentOptions.restoreSequenceCurrentValue.displayName
-			, this.deploymentOptions.rebuildIndexesOfflineForDataPhase.displayName
-			, this.deploymentOptions.preserveIdentityLastValues.displayName
-			, this.deploymentOptions.isAlwaysEncryptedParameterizationEnabled.displayName
-			, this.deploymentOptions.allowExternalLibraryPaths.displayName
-			, this.deploymentOptions.allowExternalLanguagePaths.displayName
-			, this.deploymentOptions.hashObjectNamesInLogs.displayName
-			, this.deploymentOptions.doNotDropWorkloadClassifiers.displayName
-			, this.deploymentOptions.ignoreWorkloadClassifiers.displayName
-			, this.deploymentOptions.ignoreDatabaseWorkloadGroups.displayName
-			, this.deploymentOptions.doNotDropDatabaseWorkloadGroups.displayName].sort();
+	public UpdateOptionsMapTable() {
+		this.optionsMapTable = this.deploymentOptions.optionsMapTable;
 	}
 
 	/**
-	 * Gets the options checkbox check value
-	 * @returns string[][]
+	 * Gets the options checkbox values by iterating through the labels and gets the default/changed value from the optionsMapTable
+	 * Returns data as [optionName, booleanValue], where these values are sending to the options table component
 	 */
 	public getOptionsData(): string[][] {
 		let data: any = [];
-		this.optionsLookup = {};
+		this.optionsLookup = new Map<string, boolean>();
 		this.optionsLabels.forEach(l => {
-			let checked: boolean = this.getDeployOptionUtil(l);
-			data.push([checked, l]);
-			this.optionsLookup[l] = checked;
+			let checked: boolean | undefined = this.getDeployOptionUtil(l);
+			if (checked !== undefined) {
+				data.push([checked, l]);
+				this.optionsLookup?.set(l, checked);
```

Comment:
A function with a name `get...` should generally not be having side effects like clearing and initializing a map.

Reply:
Maybe `refresh...` would be more appropriate here.